bokeh/core/property/descriptors.py

```diff
@@ -651,21 +654,27 @@ def _get_default(self, obj):
             # this shouldn't happen because we should have checked before _get_default()
             raise RuntimeError("Bokeh internal error, does not handle the case of self.name already in _property_values")
 
-        # merely getting a default may force us to put it in
-        # _property_values if we need to wrap the container, if
-        # the default is a Model that may change out from
-        # underneath us, or if the default is generated anew each
-        # time by a function.
+        is_themed = obj.themed_values() is not None and self.name in obj.themed_values()
+
         default = self.instance_default(obj)
```

Comment:
This is the only bit I am not too happy about. The value returned by `instance_default` already has a theme applied (or not) so the only way to check if it is actually a themed value is to peek into `themed_values` above. I guess `instance_default` could return `(value, is_themed)` but that seems a bit ugly.